### DISCUSSION TITLE
Is it possible to use separate packages from nextjs?

### DISCUSSION THREAD

**Discussion by ryankauk** [vercel/next.js#21372] [Help] [answered]
Hey there, I'm involved in transitioning into nextjs but I need to do some incremental adoption. There's a few utilities in next-server that I would like to use without the full overhead of nextjs right now. Is there anyway to do this? I noticed there's a next-server package but hasn't been published in a year. I'm assuming there was a decision to keep it fully integrated into the "next" package. Let me know if this is at all possible! Thank you.

**Comment by jamesmosier** [comment, marked as answer]
Hi @ryankauk. Using that package (or any other package that Next uses internally) is not publicly supported, so you are correct in your assumption that the `next` package is the only way to run a stable Next.js app. I'm not sure why that `next-server` package is npm, but it is not supported/recommended as an incremental adoption path.

There are some other options in [this blog post](https://nextjs.org/blog/incremental-adoption).

**Reply by ryankauk** [reply]
Alright thank you for the info
